Feature/parse min max

ventura-rivera opened the pull request:
### Summary
* adding support to recognize a pattern for min/max values

ventura-rivera commented:
travis-ci is not able to download python 3.3
Also travis-ci is not able to create a conda environment in python 3.4

ventura-rivera commented:
@matSciMalcolm, I cannot add you as a reviewer for some reason.

matSciMalcolm commented:
Implementation seems reasonable to me. Is there a way a user can spec a value and a min max?

ventura-rivera commented:
@matSciMalcolm Yes, this can be implemented, but we should agree on what that format should look like

maxhutch commented on csv_template_ingester/template_csv_parser.py:
```diff
@@ -548,8 +550,16 @@ def add_property(systm, property_value, names, units, column_index):
+        maximum = float(min_max["max"])
+        if minimum > maximum:
+            raise ValueError("Minimum ({}) cannot be greater than maximum ({})".format(minimum, maximum))
+        prop.scalars = Scalar(minimum=minimum, maximum=maximum)
```
`listify` returns a list, so this scalar should be wrapped in a list as well.

ventura-rivera replied:
Good point, I'll return a list

maxhutch commented on requirements.txt:
```diff
@@ -1,2 +1,3 @@
 pypif==2.1.0
 xlrd
+pytest
```
This would typically go in a `test_requirements.txt` file, since its not an install requirement.

ventura-rivera replied:
I can do that

maxhutch commented on setup.py:
```diff
@@ -9,7 +9,8 @@
         'pypif>=2.1.0,<3',
-        'xlrd'
+        'xlrd',
+        'pytest'
```
This should go in `tests_require` instead of `install_requires`, but it is usually omitted when tests aren't run through `python ./setup.py test` (which may be deprecated).

ventura-rivera replied:
I'll just omit pytest from here

ventura-rivera commented:
> is `range` case-sensitive?

@maxhutch it was before, but I have made it case insensitive (and added that test, as well)

I addressed everything else you commented on. Let me know if there was anything I missed out.
